### PR TITLE
netease-cloud-music-gtk: 1.2.2 -> 2.0.1

### LINKED ISSUE #185428: netease-cloud-music-gtk is outdated
###### Checklist

- [X] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name:

current version: `1.2.2`
desired version: `2.0.1`

###### Notify maintainers

maintainers: @Diffumist 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/185428

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/netease-cloud-music-gtk/default.nix [hunk 1]
```diff
@@ -46,15 +63,11 @@ rustPlatform.buildRustPackage rec {
     gst-plugins-ugly
   ]);
 
-  postPatch = ''
-    install -D netease-cloud-music-gtk.desktop $out/share/applications/netease-cloud-music-gtk.desktop
-    install -D icons/netease-cloud-music-gtk.svg $out/share/icons/hicolor/scalable/apps/netease-cloud-music-gtk.svg
-  '';
-
   meta = with lib; {
     description = "netease-cloud-music-gtk is a Rust + GTK based netease cloud music player";
```
Comment (jian-lin) [inline pkgs/applications/audio/netease-cloud-music-gtk/default.nix:67]:
```suggestion
    description = "A Rust + GTK based netease cloud music player";
```

**Review by jian-lin — COMMENTED** [review]
LGTM. Could you also change [this](https://github.com/NixOS/nixpkgs/pull/145320#discussion_r746447512)?

File: pkgs/applications/audio/netease-cloud-music-gtk/default.nix [hunk 2]
```diff
@@ -1,43 +1,60 @@
...
   src = fetchFromGitHub {
     owner = "gmg137";
-    repo = "netease-cloud-music-gtk";
+    repo = pname;
     rev = version;
-    sha256 = "sha256-42MaylfG5LY+TiYHWQMoh9CiVLShKXSBpMrxdWhujow=";
+    hash = "sha256-dlJZvmfw9+cavAysxVzCekgPdygg5zbU3ZR5BOjPk08=";
+  };
+
+  patches = [ ./cargo-lock.patch ];
```
Comment (jian-lin) [inline pkgs/applications/audio/netease-cloud-music-gtk/default.nix:31]: Is it better to let the upstream provide `Cargo.lock`?

Comment (NickCao) [reply pkgs/applications/audio/netease-cloud-music-gtk/default.nix:31]: Upstream refused to do so: https://github.com/gmg137/netease-cloud-music-gtk/issues/118

**Comment by bobby285271** [comment]
@ofborg build netease-cloud-music-gtk

**Pull request merged**